tests/src/tests/spore.rs

```diff
@@ -102,291 +141,401 @@ fn test_multi_spores_mint() {
+        let agent_action = if add_agent_action {
+            vec![(agent_type, action)]
+        } else {
+            vec![]
+        };
+        let mut tx = build_single_spore_mint_tx_with_extra_action(
+            &mut context,
+            "Hello Spore!".as_bytes().to_vec(),
+            "plain/text",
+            None,
+            Some(cluster_id),
+            agent_action,
+        );
+        tx = tx
+            .as_advanced_builder()
+            .input(agent_input)
+            .output(agent_output)
+            .output_data(proxy_type_hash.to_vec().pack())
+            .cell_dep(agent_script_dep)
+            .build();
+        if add_cluster_dep {
+            tx = tx.as_advanced_builder().cell_dep(cluster_dep).build();
+        }
+        tx = context.complete_tx(tx);
+
+        context
+            .verify_tx(&tx, MAX_CYCLES)
+            .expect("test spore mint from cluster transfer");
+    }
 
-    context
-        .verify_tx(&tx, MAX_CYCLES)
-        .expect("try destroy immortal");
+    #[test]
+    fn test_spore_mint_from_agent_transfer() {
+        make_spore_mint_from_agent_transfer(true, true);
+    }
+
+    #[should_panic]
+    #[test]
+    fn test_spore_mint_from_agent_transfer_failed_with_no_cluster_dep() {
+        make_spore_mint_from_agent_transfer(false, true);
+    }
+
+    #[should_panic]
+    #[test]
+    fn test_spore_mint_from_agent_transfer_failed_with_no_agent_action() {
+        make_spore_mint_from_agent_transfer(true, false);
+    }
 }
 
-#[should_panic]
-#[test]
-fn test_simple_spore_mint_failed_with_error_type() {
-    let error_nft_types = ["plain/;", "text", ";", "-", "plain/", "plain/test;;test=;"];
+mod simple_spore_destroy {
+    use super::*;
 
-    let mut context = Context::default();
-    for content_type in error_nft_types {
-        let tx = build_single_spore_mint_tx(
+    fn make_simple_spore_destroy(content_type: &str) {
+        let mut context = Context::default();
+        let serialized =
+            build_serialized_spore_data("Hello Spore!".as_bytes().to_vec(), content_type, None);
+        let (spore_out_point, spore_script_dep) =
+            build_spore_contract_materials(&mut context, "spore");
+
+        let normal_input = build_normal_input(&mut context);
+        let spore_type_id = build_type_id(&normal_input, 0);
+        let type_ = build_spore_type_script(
             &mut context,
-            "THIS IS A TEST NFT".as_bytes().to_vec(),
-            content_type,
-            None,
-            None,
+            &spore_out_point,
+            spore_type_id.to_vec().into(),
         );
+        let spore_input = build_spore_input(&mut context, type_.clone(), serialized.clone());
+
+        let normal_output = build_normal_output(&mut context);
+        let tx = TransactionBuilder::default()
+            .input(spore_input)
+            .output(normal_output)
+            .output_data(serialized.as_slice().pack())
```

Comment:
```suggestion
            .output_data(Default::default())
```

destory the spore data too.